test/optional-validation.test.js

```diff
@@ -0,0 +1,504 @@
+// This test written in mocha+should.js
+var should = require('./init.js');
+var async = require('async');
+var db, User, options, filter;
+var j = require('../');
+var ValidationError = j.ValidationError;
+
+describe('optional-validation', function () {
+
+  before(function (done) {
+    db = getSchema();
+    User = db.define('User', {
+      seq: {type: Number, index: true},
+      name: {type: String, index: true, sort: true},
+      email: {type: String, index: true},
+      birthday: {type: Date, index: true},
+      role: {type: String, index: true},
+      order: {type: Number, index: true, sort: true},
+      vip: {type: Boolean}
+    });
+
+    options = {};
+    filter = {fields: ['name', 'id']};
+
+    db.automigrate(['User'], done);
+
+  });
+
+  beforeEach(function (done) {
+    User.destroyAll(function () {
+      delete User.validations;
+      done();
+    });
+  });
+
+  describe('no model setting', function () {
+
+    it('should throw on create with validate:true with invalid data', function (done) {
+      var options = {validate: true};
+      User.validatesPresenceOf('name');
```

Comment:
Please move this line to `beforeEach` and/or top-level model setup code, so that it is not duplicated in every test. Unless I missed something?